Python/Python-tokenize.c

```diff
@@ -243,11 +247,7 @@ tokenizeriter_next(tokenizeriterobject *it)
     }
     if (it->done || type == ERRORTOKEN) {
         PyErr_SetString(PyExc_StopIteration, "EOF");
-
-        Py_BEGIN_CRITICAL_SECTION(it);
-        it->done = 1;
-        Py_END_CRITICAL_SECTION();
-
+        _Py_atomic_store_int(&it->done, 1);
```

Comment:
Might want to guard with an `ifdef Py_GIL_DISABLED` and keep the original version around if it's perf sensitive.